lib/common/util/exceptions.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import moment from 'moment'
+
+import type {ParsedRangesAndDates} from '../../types'
+
+// Method to filter out new dates (0's) so that flow is happy
+// We have to use a reducer to keep flow happy.
+export function filterOutNewDates (dates: Array<string | 0>): Array<string> {
+  const outputDates: Array<string> = []
+  const filter = (acc, current) => {
+    typeof current === 'string' && acc.push(current)
+    return acc
+  }
+  dates.reduce(filter, outputDates)
+  return outputDates
+}
+
+/**
+ * This method adds a range of dates from startRange to endRange to the dates array, inclusively according to startInclusive
+ * and endInclusive.
+ */
+export function addRangeOfDates (
+  startRange: moment,
+  endRange: moment,
+  dates: Array<string>,
+  startInclusive: boolean,
+  endInclusive: boolean
+) {
+  const datesToAdd = []
+  while (endRange.unix() >= startRange.unix()) {
+    if (endRange.unix() === startRange.unix() && !endInclusive) break
+    if (datesToAdd.length === 0 && !startInclusive) {
+      startInclusive = true
+      continue
+    }
+    datesToAdd.push(startRange.format('YYYYMMDD'))
+    startRange.add(1, 'days')
+  }
+  datesToAdd.forEach(date => {
+    if (!dates.some(el => el === date)) {
+      dates.push(date)
+    }
+  })
+  return dates
+}
+
+/**
+ * This method removes a range of dates from startRange to endRange to the dates array, inclusively according to startInclusive
+ * and endInclusive.
+ */
+export function removeRangeOfDates (
```

Comment:
Like `addRangeOfDates`, I would make this a recursive method which should reduce complexity and the number of arguments needed. Since we have a pair of methods now, we could even add a flag to those methods to switch them between add and remove modes

Reply:
combined these methods and made recursive in https://github.com/ibi-group/datatools-ui/pull/828/commits/0bd970887b85994ed9f9452ec5640b01eb021f28